engine/execution/messages.go

```diff
@@ -50,3 +50,11 @@ func (cr *ComputationResult) AddComputationUsed(inp uint64) {
 func (cr *ComputationResult) AddStateSnapshot(inp *delta.SpockSnapshot) {
 	cr.StateSnapshots = append(cr.StateSnapshots, inp)
 }
+
+func (cr *ComputationResult) GetBlockID() *flow.Identifier {
+	if cr.ExecutableBlock != nil && cr.ExecutableBlock.Block != nil {
+		availBlockID := cr.ExecutableBlock.Block.ID()
```

Comment:
the purpose of this function is simply to handle the case when `cr.ExecutableBlock` or `cr.ExecutableBlock.Block` is nil, because both of them are pointers. If they will never be `nil`, I agree we will never need it, however they expose risks that in future changes they might be initialized\set to nil.

Reply:
discussed offline - i will simply take the existing assumption that `computationResult.ExecutableBlock` will never be `nil`. Now this PR is much simpler.